xcb features:  method `from_glib_none` is not a member of trait `FromGlibPtrFull`

According to the [doc](http://gtk-rs.org/docs/glib/translate/trait.FromGlibPtrFull.html?search=FromGlibPtr), we have now multiples traits for each method. Here is the error after putting this in my ```Cargo.toml```:
```
[dependencies.cairo-rs]
version = "0.1.3"
default-features = false
features = ["xcb"]
```
```
error[E0407]: method `from_glib_none` is not a member of trait `FromGlibPtrFull`
  --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:42:5
   |
42 |       unsafe fn from_glib_none(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
   |  _____^ starting here...
43 | |         assert!(!ptr.is_null());
44 | |         XCBConnection(ptr)
45 | |     }
   | |_____^ ...ending here: not a member of trait `FromGlibPtrFull`

error[E0407]: method `from_glib_none` is not a member of trait `FromGlibPtrFull`
  --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:80:5
   |
80 |       unsafe fn from_glib_none(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
   |  _____^ starting here...
81 | |         assert!(!ptr.is_null());
82 | |         XCBRenderPictFormInfo(ptr)
83 | |     }
   | |_____^ ...ending here: not a member of trait `FromGlibPtrFull`

error[E0277]: the trait bound `xcb::XCBConnection: glib::translate::FromGlibPtrNone<*mut ffi::xcb_connection_t>` is not satisfied
  --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:62:18
   |
62 |         unsafe { from_glib_none(self.to_glib_none().0) }
   |                  ^^^^^^^^^^^^^^ the trait `glib::translate::FromGlibPtrNone<*mut ffi::xcb_connection_t>` is not implemented for `xcb::XCBConnection`
   |
   = note: required by `glib::translate::from_glib_none`

error[E0277]: the trait bound `xcb::XCBRenderPictFormInfo: glib::translate::FromGlibPtrNone<*mut ffi::xcb_render_pictforminfo_t>` is not satisfied
   --> /home/charles/.cargo/registry/src/github.com-1ecc6299db9ec823/cairo-rs-0.1.3/src/xcb.rs:100:18
    |
100 |         unsafe { from_glib_none(self.to_glib_none().0) }
    |                  ^^^^^^^^^^^^^^ the trait `glib::translate::FromGlibPtrNone<*mut ffi::xcb_render_pictforminfo_t>` is not implemented for `xcb::XCBRenderPictFormInfo`
    |
    = note: required by `glib::translate::from_glib_none`

error: aborting due to 2 previous errors

error: Could not compile `cairo-rs`.
Build failed, waiting for other jobs to finish...
error: build failed
```

--- a/src/xcb.rs
+++ b/src/xcb.rs
@@ -2,6 +2,7 @@
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
+#[cfg(feature = "glib")]
 use glib::translate::*;
 use ffi;
 
@@ -11,7 +12,7 @@ use surface::Surface;
 pub struct XCBDrawable(pub u32);
 
 impl XCBDrawable {
-    fn to_glib_none(&self) -> u32 {
+    fn to_raw_none(&self) -> u32 {
         self.0
     }
 }
@@ -20,7 +21,7 @@ impl XCBDrawable {
 pub struct XCBPixmap(pub u32);
 
 impl XCBPixmap {
-    fn to_glib_none(&self) -> u32 {
+    fn to_raw_none(&self) -> u32 {
         self.0
     }
 }
@@ -28,28 +29,45 @@ impl XCBPixmap {
 
 pub struct XCBConnection(pub *mut ffi::xcb_connection_t);
 
+impl XCBConnection {
+    pub fn to_raw_none(&self) -> *mut ffi::xcb_connection_t {
+        self.0
+    }
+
+    pub unsafe fn from_raw_none(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
+        assert!(!ptr.is_null());
+        XCBConnection(ptr)
+    }
+
+    pub unsafe fn from_raw_full(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
+        assert!(!ptr.is_null());
+        XCBConnection(ptr)
+    }
+}
+
+#[cfg(feature = "glib")]
 impl<'a> ToGlibPtr<'a, *mut ffi::xcb_connection_t> for &'a XCBConnection {
     type Storage = &'a XCBConnection;
 
     #[inline]
     fn to_glib_none(&self) -> Stash<'a, *mut ffi::xcb_connection_t, &'a XCBConnection> {
-        Stash(self.0, *self)
+        Stash(self.to_raw_none(), *self)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrNone<*mut ffi::xcb_connection_t> for XCBConnection {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
-        assert!(!ptr.is_null());
-        XCBConnection(ptr)
+        Self::from_raw_none(ptr)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrFull<*mut ffi::xcb_connection_t> for XCBConnection {
     #[inline]
     unsafe fn from_glib_full(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
-        assert!(!ptr.is_null());
-        XCBConnection(ptr)
+        Self::from_raw_full(ptr)
     }
 }
 
@@ -61,35 +79,52 @@ impl AsRef<XCBConnection> for XCBConnection {
 
 impl Clone for XCBConnection {
     fn clone(&self) -> XCBConnection {
-        unsafe { from_glib_none(self.to_glib_none().0) }
+        unsafe { Self::from_raw_none(self.to_raw_none()) }
     }
 }
 
 
 pub struct XCBRenderPictFormInfo(pub *mut ffi::xcb_render_pictforminfo_t);
 
+impl XCBRenderPictFormInfo {
+    pub fn to_raw_none(&self) ->  *mut ffi::xcb_render_pictforminfo_t {
+        self.0
+    }
+
+    pub unsafe fn from_raw_none(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
+        assert!(!ptr.is_null());
+        XCBRenderPictFormInfo(ptr)
+    }
+
+    pub unsafe fn from_raw_full(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
+        assert!(!ptr.is_null());
+        XCBRenderPictFormInfo(ptr)
+    }
+}
+
+#[cfg(feature = "glib")]
 impl<'a> ToGlibPtr<'a, *mut ffi::xcb_render_pictforminfo_t> for &'a XCBRenderPictFormInfo {
     type Storage = &'a XCBRenderPictFormInfo;
 
     #[inline]
     fn to_glib_none(&self) -> Stash<'a, *mut ffi::xcb_render_pictforminfo_t, &'a XCBRenderPictFormInfo> {
-        Stash(self.0, *self)
+        Stash(self.to_raw_none(), *self)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrNone<*mut ffi::xcb_render_pictforminfo_t> for XCBRenderPictFormInfo {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
-        assert!(!ptr.is_null());
-        XCBRenderPictFormInfo(ptr)
+        Self::from_raw_none(ptr)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrFull<*mut ffi::xcb_render_pictforminfo_t> for XCBRenderPictFormInfo {
     #[inline]
     unsafe fn from_glib_full(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
-        assert!(!ptr.is_null());
-        XCBRenderPictFormInfo(ptr)
+        Self::from_raw_full(ptr)
     }
 }
 
@@ -101,35 +136,52 @@ impl AsRef<XCBRenderPictFormInfo> for XCBRenderPictFormInfo {
 
 impl Clone for XCBRenderPictFormInfo {
     fn clone(&self) -> XCBRenderPictFormInfo {
-        unsafe { from_glib_none(self.to_glib_none().0) }
+        unsafe { Self::from_raw_none(self.to_raw_none()) }
     }
 }
 
 
 pub struct XCBScreen(pub *mut ffi::xcb_screen_t);
 
+impl XCBScreen {
+    pub fn to_raw_none(&self) -> *mut ffi::xcb_screen_t {
+        self.0
+    }
+
+    pub unsafe fn from_raw_none(ptr: *mut ffi::xcb_screen_t) -> XCBScreen {
+        assert!(!ptr.is_null());
+        XCBScreen(ptr)
+    }
+
+    pub unsafe fn from_raw_full(ptr: *mut ffi::xcb_screen_t) -> XCBScreen {
+        assert!(!ptr.is_null());
+        XCBScreen(ptr)
+    }
+}
+
+#[cfg(feature = "glib")]
 impl<'a> ToGlibPtr<'a, *mut ffi::xcb_screen_t> for &'a XCBScreen {
     type Storage = &'a XCBScreen;
 
     #[inline]
     fn to_glib_none(&self) -> Stash<'a, *mut ffi::xcb_screen_t, &'a XCBScreen> {
-        Stash(self.0, *self)
+        Stash(self.to_raw_none(), *self)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrNone<*mut ffi::xcb_screen_t> for XCBScreen {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut ffi::xcb_screen_t) -> XCBScreen {
-        assert!(!ptr.is_null());
-        XCBScreen(ptr)
+        Self::from_raw_none(ptr)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrFull<*mut ffi::xcb_screen_t> for XCBScreen {
     #[inline]
     unsafe fn from_glib_full(ptr: *mut ffi::xcb_screen_t) -> XCBScreen {
-        assert!(!ptr.is_null());
-        XCBScreen(ptr)
+        Self::from_raw_full(ptr)
     }
 }
 
@@ -141,7 +193,7 @@ impl AsRef<XCBScreen> for XCBScreen {
 
 impl Clone for XCBScreen {
     fn clone(&self) -> XCBScreen {
-        unsafe { from_glib_none(self.to_glib_none().0) }
+        unsafe { Self::from_raw_none(self.to_raw_none()) }
     }
 }
 
@@ -162,22 +214,22 @@ impl XCBSurface for Surface {
     fn create(connection: &XCBConnection, drawable: &XCBDrawable, visual: &XCBVisualType,
               width: i32, height: i32) -> Surface {
         unsafe {
-            from_glib_full(ffi::cairo_xcb_surface_create(connection.to_glib_none().0,
-                                                         drawable.to_glib_none(),
-                                                         visual.to_glib_none().0,
-                                                         width,
-                                                         height))
+            Surface::from_raw_full(ffi::cairo_xcb_surface_create(connection.to_raw_none(),
+                                                                 drawable.to_raw_none(),
+                                                                 visual.to_raw_none(),
+                                                                 width,
+                                                                 height))
         }
     }
 
     fn create_for_bitmap(connection: &XCBConnection, screen: &XCBScreen, bitmap: &XCBPixmap,
                          width: i32, height: i32) -> Surface {
         unsafe {
-            from_glib_full(ffi::cairo_xcb_surface_create_for_bitmap(connection.to_glib_none().0,
-                                                                    screen.to_glib_none().0,
-                                                                    bitmap.to_glib_none(),
-                                                                    width,
-                                                                    height))
+            Surface::from_raw_full(ffi::cairo_xcb_surface_create_for_bitmap(connection.to_raw_none(),
+                                                                            screen.to_raw_none(),
+                                                                            bitmap.to_raw_none(),
+                                                                            width,
+                                                                            height))
         }
     }
 
@@ -185,24 +237,24 @@ impl XCBSurface for Surface {
                                   bitmap: &XCBPixmap, format: &XCBRenderPictFormInfo,
                                   width: i32, height: i32) -> Surface {
         unsafe {
-            from_glib_full(ffi::cairo_xcb_surface_create_with_xrender_format(connection.to_glib_none().0,
-                                                                             screen.to_glib_none().0,
-                                                                             bitmap.to_glib_none(),
-                                                                             format.to_glib_none().0,
-                                                                             width,
-                                                                             height))
+            Surface::from_raw_full(ffi::cairo_xcb_surface_create_with_xrender_format(connection.to_raw_none(),
+                                                                                     screen.to_raw_none(),
+                                                                                     bitmap.to_raw_none(),
+                                                                                     format.to_raw_none(),
+                                                                                     width,
+                                                                                     height))
         }
     }
 
     fn set_size(self: &Surface, width: i32, height: i32) {
         unsafe {
-            ffi::cairo_xcb_surface_set_size(self.to_glib_none().0, width, height)
+            ffi::cairo_xcb_surface_set_size(self.to_raw_none(), width, height)
         }
     }
 
     fn set_drawable(self: &Surface, drawable: &XCBDrawable, width: i32, height: i32) {
         unsafe {
-            ffi::cairo_xcb_surface_set_drawable(self.to_glib_none().0, drawable.to_glib_none(),
+            ffi::cairo_xcb_surface_set_drawable(self.to_raw_none(), drawable.to_raw_none(),
                                                 width, height)
         }
     }
@@ -211,28 +263,45 @@ impl XCBSurface for Surface {
 
 pub struct XCBVisualType(pub *mut ffi::xcb_visualtype_t);
 
+impl XCBVisualType {
+    pub fn to_raw_none(&self) -> *mut ffi::xcb_visualtype_t {
+        self.0
+    }
+
+    pub unsafe fn from_raw_none(ptr: *mut ffi::xcb_visualtype_t) -> XCBVisualType {
+        assert!(!ptr.is_null());
+        XCBVisualType(ptr)
+    }
+
+    pub unsafe fn from_raw_full(ptr: *mut ffi::xcb_visualtype_t) -> XCBVisualType {
+        assert!(!ptr.is_null());
+        XCBVisualType(ptr)
+    }
+}
+
+#[cfg(feature = "glib")]
 impl<'a> ToGlibPtr<'a, *mut ffi::xcb_visualtype_t> for &'a XCBVisualType {
     type Storage = &'a XCBVisualType;
 
     #[inline]
     fn to_glib_none(&self) -> Stash<'a, *mut ffi::xcb_visualtype_t, &'a XCBVisualType> {
-        Stash(self.0, *self)
+        Stash(self.to_raw_none(), *self)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrNone<*mut ffi::xcb_visualtype_t> for XCBVisualType {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut ffi::xcb_visualtype_t) -> XCBVisualType {
-        assert!(!ptr.is_null());
-        XCBVisualType(ptr)
+        Self::from_raw_none(ptr)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrFull<*mut ffi::xcb_visualtype_t> for XCBVisualType {
     #[inline]
     unsafe fn from_glib_full(ptr: *mut ffi::xcb_visualtype_t) -> XCBVisualType {
-        assert!(!ptr.is_null());
-        XCBVisualType(ptr)
+        Self::from_raw_full(ptr)
     }
 }
 
@@ -244,35 +313,52 @@ impl AsRef<XCBVisualType> for XCBVisualType {
 
 impl Clone for XCBVisualType {
     fn clone(&self) -> XCBVisualType {
-        unsafe { from_glib_none(self.to_glib_none().0) }
+        unsafe { Self::from_raw_none(self.to_raw_none()) }
     }
 }
 
 
 pub struct Device(pub *mut ffi::cairo_device_t);
 
+impl Device {
+    pub fn to_raw_none(&self) -> *mut ffi::cairo_device_t {
+        self.0
+    }
+
+    pub unsafe fn from_raw_none(ptr: *mut ffi::cairo_device_t) -> Device {
+        assert!(!ptr.is_null());
+        Device(ptr)
+    }
+
+    pub unsafe fn from_raw_full(ptr: *mut ffi::cairo_device_t) -> Device {
+        assert!(!ptr.is_null());
+        Device(ptr)
+    }
+}
+
+#[cfg(feature = "glib")]
 impl<'a> ToGlibPtr<'a, *mut ffi::cairo_device_t> for &'a Device {
     type Storage = &'a Device;
 
     #[inline]
     fn to_glib_none(&self) -> Stash<'a, *mut ffi::cairo_device_t, &'a Device> {
-        Stash(self.0, *self)
+        Stash(self.to_raw_none(), *self)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrNone<*mut ffi::cairo_device_t> for Device {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut ffi::cairo_device_t) -> Device {
-        assert!(!ptr.is_null());
-        Device(ptr)
+        Self::from_raw_none(ptr)
     }
 }
 
+#[cfg(feature = "glib")]
 impl FromGlibPtrFull<*mut ffi::cairo_device_t> for Device {
     #[inline]
     unsafe fn from_glib_full(ptr: *mut ffi::cairo_device_t) -> Device {
-        assert!(!ptr.is_null());
-        Device(ptr)
+        Self::from_raw_full(ptr)
     }
 }
 
@@ -284,20 +370,20 @@ impl AsRef<Device> for Device {
 
 impl Clone for Device {
     fn clone(&self) -> Device {
-        unsafe { from_glib_none(self.to_glib_none().0) }
+        unsafe { Self::from_raw_none(self.to_raw_none()) }
     }
 }
 
 impl Device {
     pub fn get_connection(&self) -> XCBConnection {
         unsafe {
-            from_glib_full(ffi::cairo_xcb_device_get_connection(self.to_glib_none().0))
+            XCBConnection::from_raw_full(ffi::cairo_xcb_device_get_connection(self.to_raw_none()))
         }
     }
 
     pub fn debug_cap_xrender_version(&self, major_version: i32, minor_version: i32) {
         unsafe {
-            ffi::cairo_xcb_device_debug_cap_xrender_version(self.to_glib_none().0,
+            ffi::cairo_xcb_device_debug_cap_xrender_version(self.to_raw_none(),
                                                             major_version,
                                                             minor_version)
         }
@@ -305,7 +391,7 @@ impl Device {
 
     pub fn debug_cap_xshm_version(&self, major_version: i32, minor_version: i32) {
         unsafe {
-            ffi::cairo_xcb_device_debug_cap_xshm_version(self.to_glib_none().0,
+            ffi::cairo_xcb_device_debug_cap_xshm_version(self.to_raw_none(),
                                                          major_version,
                                                          minor_version)
         }
@@ -313,13 +399,13 @@ impl Device {
 
     pub fn debug_get_precision(&self) -> i32 {
         unsafe {
-            ffi::cairo_xcb_device_debug_get_precision(self.to_glib_none().0)
+            ffi::cairo_xcb_device_debug_get_precision(self.to_raw_none())
         }
     }
 
     pub fn debug_set_precision(&self, precision: i32) {
         unsafe {
-            ffi::cairo_xcb_device_debug_set_precision(self.to_glib_none().0, precision)
+            ffi::cairo_xcb_device_debug_set_precision(self.to_raw_none(), precision)
         }
     }
 }

--- a/src/xcb.rs
+++ b/src/xcb.rs
@@ -37,13 +37,15 @@ impl<'a> ToGlibPtr<'a, *mut ffi::xcb_connection_t> for &'a XCBConnection {
     }
 }
 
-impl FromGlibPtrFull<*mut ffi::xcb_connection_t> for XCBConnection {
+impl FromGlibPtrNone<*mut ffi::xcb_connection_t> for XCBConnection {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
         assert!(!ptr.is_null());
         XCBConnection(ptr)
     }
+}
 
+impl FromGlibPtrFull<*mut ffi::xcb_connection_t> for XCBConnection {
     #[inline]
     unsafe fn from_glib_full(ptr: *mut ffi::xcb_connection_t) -> XCBConnection {
         assert!(!ptr.is_null());
@@ -75,13 +77,15 @@ impl<'a> ToGlibPtr<'a, *mut ffi::xcb_render_pictforminfo_t> for &'a XCBRenderPic
     }
 }
 
-impl FromGlibPtrFull<*mut ffi::xcb_render_pictforminfo_t> for XCBRenderPictFormInfo {
+impl FromGlibPtrNone<*mut ffi::xcb_render_pictforminfo_t> for XCBRenderPictFormInfo {
     #[inline]
     unsafe fn from_glib_none(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
         assert!(!ptr.is_null());
         XCBRenderPictFormInfo(ptr)
     }
+}
 
+impl FromGlibPtrFull<*mut ffi::xcb_render_pictforminfo_t> for XCBRenderPictFormInfo {
     #[inline]
     unsafe fn from_glib_full(ptr: *mut ffi::xcb_render_pictforminfo_t) -> XCBRenderPictFormInfo {
         assert!(!ptr.is_null());
@@ -147,7 +151,7 @@ pub trait XCBSurface {
               width: i32, height: i32) -> Surface;
     fn create_for_bitmap(connection: &XCBConnection, screen: &XCBScreen, bitmap: &XCBPixmap,
                          width: i32, height: i32) -> Surface;
-    fn create_with_xrender_format(connection: &XCBConnection, screen: &XCBScreen, 
+    fn create_with_xrender_format(connection: &XCBConnection, screen: &XCBScreen,
                                   bitmap: &XCBPixmap, format: &XCBRenderPictFormInfo,
                                   width: i32, height: i32) -> Surface;
     fn set_size(&self, width: i32, height: i32);
@@ -177,7 +181,7 @@ impl XCBSurface for Surface {
         }
     }
 
-    fn create_with_xrender_format(connection: &XCBConnection, screen: &XCBScreen, 
+    fn create_with_xrender_format(connection: &XCBConnection, screen: &XCBScreen,
                                   bitmap: &XCBPixmap, format: &XCBRenderPictFormInfo,
                                   width: i32, height: i32) -> Surface {
         unsafe {

